src/local.py

```diff
@@ -1,84 +1,113 @@
+import os, logging
 
-import sys
-if sys.platform == 'win32':
+from consts import (
+    GameID,
+    REGISTRY_START_PATHS,
+    SOFTWARE_PATHS,
+    GAME_REGISTY_RELATIVE_LOCATIONS,
+    REGISTRY_EXE_KEYS,
+    WIN_UNINSTALL_RELATIVE_LOCATION,
+    IS_WINDOWS,
+)
+import utils
+
+
+if IS_WINDOWS:
     import winreg
 
-from consts import MINECRAFT_REGISTRY_PATHS, REGISTRY_START_PATHS, MINECRAFT_REGISTRY_PATH_INSTALL_LOCATION_KEY, WINDOWS_UNINSTALL_LOCATION
+log = logging.getLogger(__name__)
 
-import logging as log
-import os
-import psutil
-import asyncio
 
-class LocalClient():
+class LocalClient:
     def __init__(self):
-        self.running_process = None
-
-    async def get_size_at_path(self, start_path):
-        total_size = 0
-        for dirpath, dirnames, filenames in os.walk(start_path):
-            for f in filenames:
-                await asyncio.sleep(0)
-                fp = os.path.join(dirpath, f)
-                if not os.path.islink(fp):
-                    total_size += os.path.getsize(fp)
-
-        return total_size
-
-    def get_minecraft_launcher_path(self):
-        if sys.platform == 'win32':
-            install_path = self.get_minecraft_folder_path()
-            potential_path = os.path.join(install_path, 'MinecraftLauncher.exe')
-            if os.path.exists(potential_path):
-                return potential_path
-        else:
-            return self.get_minecraft_folder_path()
-
-    def get_minecraft_folder_path(self):
-        if sys.platform == 'win32':
-            for start_path in REGISTRY_START_PATHS:
-                for minecraft_reg_path in MINECRAFT_REGISTRY_PATHS:
-                    try:
-                        reg = winreg.ConnectRegistry(None, start_path)
-                        with winreg.OpenKey(reg, minecraft_reg_path) as key:
-                            install_path = winreg.QueryValueEx(key, MINECRAFT_REGISTRY_PATH_INSTALL_LOCATION_KEY)[0]
-                    except OSError:
-                        continue
-                    if os.path.exists(install_path):
-                        return install_path
-        else:
-            potential_path = "/Applications/Minecraft.app"
-            if os.path.exists(potential_path):
-                return potential_path
+        self.running_games = {
+            GameID.Minecraft: None,
+            GameID.MinecraftDungeons: None,
+        }
+
+    def find_launcher_path(self, game_id, *, folder=False, folder_path=None, exe=None) -> str:
+        if exe is None and folder_path is not None:
+            folder_path = None
+        if exe is not None and folder_path is None:
+            folder_path = exe
+        if exe is not None and folder_path is not None:
+            folder_path = os.path.abspath(folder_path)
+            exe = os.path.abspath(exe)
+        log.debug(f"folder path for {game_id}: {folder_path}")
+        log.debug(f"path for {game_id}: {exe}")
+        return exe if not folder else folder_path

```

Comment:
Yeah I've touched on the badness of this code in this reply to one of your other comments ([here](https://github.com/FriendsOfGalaxy/galaxy-integration-minecraft/pull/8#discussion_r482821587)). Basically you are completely correct. It does work so I'm not touching it but definitely it would be better if a refactor it in the future.

Reply:
Just a tip for the future: follow single responsibility rule. Keep logic in relevant layer. Do simple  interfaces: single method that do single simple thing like: `find_launcher_dir()` (without arguments that modify the inner logic). Such method should return always path to launcher and that's it. Then you can create `find_executable(directory, [game_id](if really needed))` to get exact executable and `run()` method separate for Mac and Windows that utilize other methods under the hood.

Reply:
Thanks that's very true, I will be doing some heavy refactoring in local.py

Reply:
Please do not refactor until we merge current working state :)